app/org/sagebionetworks/bridge/dynamodb/DynamoAppConfigDao.java

```diff
@@ -116,24 +117,24 @@ public void deleteAppConfig(StudyIdentifier studyId, String guid) {
         checkNotNull(studyId);
         checkNotNull(guid);
         
-        // If app config doesn't exist, fail silently. It's more user-friendly
         AppConfig appConfig = loadAppConfig(studyId, guid);
-        if (appConfig != null) {
-            appConfig.setDeleted(true);
-            mapper.save(appConfig);
+        if (appConfig == null || appConfig.isDeleted()) {
+            throw new EntityNotFoundException(AppConfig.class);
         }
+        appConfig.setDeleted(true);
+        mapper.save(appConfig);
     }
     
     public void deleteAppConfigPermanently(StudyIdentifier studyId, String guid) {
         checkNotNull(studyId);
         checkNotNull(guid);
         
-        // If app config doesn't exist, fail silently. It's more user-friendly
         DynamoAppConfig appConfig = loadAppConfig(studyId, guid);
-        if (appConfig != null) {
-            mapper.delete(appConfig);
-            criteriaDao.deleteCriteria(appConfig.getCriteria().getKey());
+        if (appConfig == null) {
+            throw new EntityNotFoundException(AppConfig.class);
         }
```

Comment:
Unit test for delete permanently a missing app config?

Reply:
Added